### ISSUE TITLE
swapon: swapfile: Function not implemented

### ISSUE THREAD

**Issue by cryptocookies** [c-sky/forum#78]
I can't enable swap, do I need to change anything when I compile using buildroot?

**Comment by cp0613** [comment]
Yes, this is because the Linux kernel in the buildroot does not turn on **CONFIG_SWAP** by default. There is a temporary solution as follows:
1. Open buildroot/board/nationalchip/gx66xx/gx66xx.config, search for CONFIG_SWAP, can find # CONFIG_SWAP is not set, modify it to **CONFIG_SWAP=y** and save it.
2. Open buildroot/output/build/linux-4.9.56/.config, also search for CONFIG_SWAP, can find # CONFIG_SWAP is not set, modify it to **CONFIG_SWAP=y** and save it.
3. make csky_gx6605s_fbcon_br_defconfig & make

The following is my revised debugging record:
```
# free
             total       used       free     shared    buffers     cached
Mem:         58964      57148       1816         16        400      44860
-/+ buffers/cache:      11888      47076
Swap:            0          0          0
# dd if=/dev/zero of=/swapfile bs=1M count=128
128+0 records in
128+0 records out
# mkswap /swapfile
Setting up swapspace version 1, size = 134213632 bytes
# swapon /swapfile
[  649.577787] Adding 131056k swap on /swapfile.  Priority:-1 extents:3 across:151544k 
# free
             total       used       free     shared    buffers     cached
Mem:         58964      56980       1984         16        424      44652
-/+ buffers/cache:      11904      47060
Swap:       131056          0     131056
# 
```

You can refer to it [here](https://www.linuxquestions.org/questions/linux-general-1/2-6-3-swapon-function-not-implemented-155442/).

**Comment by cp0613** [comment]
By the way, if you want to configure it to boot automatically, you can:
1. Open /etc/fstab and add `/swapfile swap swap defaults 0 0`
2. Open /etc/inittab and add `::sysinit:/sbin/swapon -a` before `::sysinit:/bin/mount -a`
